### PR TITLE
Implement searching on top of filtering

### PR DESCRIPTION
**Pull request by dan-f**
## Description

This PR introduces `getSearchString` and `hasActiveSearch` by re-implementing searching on top of filtering, thus combining unnecessarily divergent code paths for consistency and to prevent future regressions.

## Post-review
- [ ] Squash commits into discrete sets of changes with descriptive commit messages.

## Reviewers

- [x] @peter-fogg 
- [x] @bjacobel  

If you've been tagged for review, please check your corresponding box once you've given the :+1:.

We'll also automatically suggest reviewers for this PR based on the code it touches.

### CODE REVIEW

File: src/js/pagination/paging-collection.js [hunk 1]
```diff
@@ -302,19 +300,23 @@
             },
 
             /**
-             * Gets an object of currently active (applied) filters.
+             * Gets an object of currently active (applied) filters excluding
+             * the active search.
              * @returns {Object} An object mapping the names of
```
Comment (bjacobel) [inline src/js/pagination/paging-collection.js]: Should update this jsdoc string to describe the new argument to this method.

**Comment by bjacobel** [comment]
LGTM after the doc fix :thumbsup:

**Comment by andy-armstrong** [comment]
👍 nice clean up! 

Would you be willing to write test to cover the last few lines from this file? It is so close to 100%!

https://coveralls.io/builds/5993304/source?filename=src%2Fjs%2Fpagination%2Fpaging-collection.js

File: src/js/pagination/paging-collection.js [hunk 2]
```diff
@@ -343,11 +345,9 @@
                     value = value.join(',');
                 }
                 this.filterableFields[fieldName].value = value;
-                if (_.isUndefined(this.queryParams.fieldName)) {
```
Comment (peter-fogg) [inline src/js/pagination/paging-collection.js]: I'm not totally clear on why the conditional is being removed here -- does something about the refactor make it unnecessary? Was it ever needed?

**Comment by peter-fogg** [comment]
👍

File: src/js/pagination/paging-collection.js (hunk 2 above)
Comment (dan-f) [inline src/js/pagination/paging-collection.js]: It wasn't ever needed; I mistakenly added it in order to not have to redefine the same function over and over. However I ran into some edge cases in my consumer code where changing the value of a filter didn't actually result in the new (rather the old) value being sent in the subsequent request. Removing this conditional fixed that issue. Without diving too deep into the code or thinking about it too hard I'm pretty sure it had to do with JS's odd closure implementation.

File: src/js/pagination/paging-collection.js (hunk 1 above)
Comment (dan-f) [inline src/js/pagination/paging-collection.js]: Good catch!

**Comment by dan-f** [comment]
@bjacobel I clarified that JSDoc comment.
@andy-armstrong I *think* that this file should be 100% covered now, but I couldn't get the coverage report to update locally. I'll re-ping you when I confirm and am seeking the final 👍  :)

**Comment by bjacobel** [comment]
LGTM :thumbsup:

**Comment by dan-f** [comment]
@andy-armstrong - coverage is now at 100%!

**Comment by andy-armstrong** [comment]
Awesome @dan-f. 👍

**Pull request merged**